### PR TITLE
Add sequenceNumber to L1Block.sol

### LINKED ISSUE #192: Update L1-info contract (the L2 predeploy) to store sequence number

### PR DESCRIPTION
**Pull request by maurelian**
**Description**

Adds `sequenceNumber` to data written in the L1 Attributed deposited tx

**Metadata**
- Fixes #192 
- Fixes ENG-1976

### CODE REVIEW

File: packages/contracts/contracts/L2/L1Block.sol [hunk 1]
```diff
@@ -30,6 +32,7 @@ contract L1Block {
         number = _number;
         timestamp = _timestamp;
         basefee = _basefee;
+        basefee = _sequenceNumber;
```
Comment (trianglesphere) [inline packages/contracts/contracts/L2/L1Block.sol:35]: ?, waiting on something else?

Comment (maurelian) [reply packages/contracts/contracts/L2/L1Block.sol:35]: typo, fixed.

**Comment by maurelian** [comment]
Leaving in draft until the specs and ref-impl land. #182

**Comment by norswap** [comment]
I'm sorry for the silly question, but what is a sequence number? Is it an epoch number? (And if so, is that what we should call it?)

Sequence number sounds like L2 block number to me, which is something that couldn't possibly be written here.

**Comment by protolambda** [comment]
@norswap it's from an earlier idea to track how many L2 blocks have past since the last L1-info-deposit was included, so users/contracts can find it easily, and we might end up using it to try find the sync starting point faster. It's not really blocking anything, but was interesting enough to list in the epic. Focusing on the other sequencer work for now.

**Comment by tynes** [comment]
Should this contract be behind a delegatecall proxy? In case it needs to be upgraded to include new properties like the value of `RANDOM`

**Comment by norswap** [comment]
@tynes Just saw this. That's one option, with the other one being to make the change part of the hard fork, which would mean increasing the geth diff.

How would you go about the hard fork stuff seems like a bit of a headache however. Do we just stuff the contract in geth's memory? (That's the easiest way to do it with a hard fork.) Proxy definitely feels much easier.

**Pull request closed without merging**